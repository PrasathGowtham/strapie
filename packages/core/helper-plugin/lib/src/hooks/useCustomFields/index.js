/**
 *
 * useCustomFields
 *
 */

import { useContext, useRef } from 'react';
import CustomFieldsContext from '../../contexts/CustomFieldsContext';

const useCustomFields = () => {
  const customFields = useContext(CustomFieldsContext);
  // Use a ref so we can safely add the custom fields to a hook dependencies array
  const customFieldsRef = useRef(customFields);

  return customFieldsRef.current;
};

export default useCustomFields;
