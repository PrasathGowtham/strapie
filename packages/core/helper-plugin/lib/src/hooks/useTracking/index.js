import { useContext, useRef } from 'react';
import axios from 'axios';
import TrackingContext from '../../contexts/TrackingContext';
import useAppInfos from '../useAppInfos';

const useTracking = () => {
  const trackRef = useRef();
  const { uuid, telemetryProperties } = useContext(TrackingContext);
  const appInfo = useAppInfos();

  trackRef.current = async (event, properties) => {
    if (uuid && !window.strapi.telemetryDisabled) {
      try {
        await axios.post('https://analytics.strapi.io/track', {
          event,
          properties: {
            ...telemetryProperties,
            ...properties,
            projectType: window.strapi.projectType,
            environment: appInfo.currentEnvironment,
          },
          uuid,
        });
      } catch (err) {
        // Silent
      }
    }
  };

  return { trackUsage: trackRef.current };
};

export default useTracking;
