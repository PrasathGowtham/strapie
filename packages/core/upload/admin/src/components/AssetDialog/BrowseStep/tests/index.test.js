import React from 'react';
import { IntlProvider } from 'react-intl';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { render, fireEvent, screen } from '@testing-library/react';
import { NotificationsProvider } from '@strapi/helper-plugin';
import { MemoryRouter } from 'react-router-dom';
import { QueryClientProvider, QueryClient } from 'react-query';

import { useFolder } from '../../../../hooks/useFolder';
import { BrowseStep } from '..';

jest.mock('../../../../hooks/useFolder');

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
}));

const FIXTURE_ASSETS = [
  {
    id: 77,
    name: '3874873.jpg',
    alternativeText: null,
    caption: null,
    width: 400,
    height: 400,
    formats: {
      thumbnail: {
        name: 'thumbnail_3874873.jpg',
        hash: 'thumbnail_3874873_b5818bb250',
        ext: '.jpg',
        mime: 'image/jpeg',
        width: 156,
        height: 156,
        size: 3.97,
        path: null,
        url: '/uploads/thumbnail_3874873_b5818bb250.jpg',
      },
    },
    hash: '3874873_b5818bb250',
    ext: '.jpg',
    mime: 'image/jpeg',
    size: 11.79,
    url: '/uploads/3874873_b5818bb250.jpg',
    previewUrl: null,
    provider: 'local',
    provider_metadata: null,
    createdAt: '2021-10-18T08:04:56.326Z',
    updatedAt: '2021-10-18T08:04:56.326Z',
  },
];

const FIXTURE_FOLDERS = [
  {
    id: 1,
    createdAt: '',
    pathId: 1,
    name: 'Folder 1',
    children: {
      count: 1,
    },
    files: {
      count: 1,
    },
    updatedAt: '',
    path: '/1',
  },
];

const client = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const ComponentFixture = (props) => {
  return (
    <QueryClientProvider client={client}>
      <ThemeProvider theme={lightTheme}>
        <MemoryRouter>
          <IntlProvider messages={{}} locale="en">
            <NotificationsProvider toggleNotification={() => {}}>
              <BrowseStep
                assets={[]}
                canCreate
                canRead
                folders={FIXTURE_FOLDERS}
                onAddAsset={jest.fn()}
                onChangeFilters={jest.fn()}
                onChangePage={jest.fn()}
                onChangePageSize={jest.fn()}
                onChangeSearch={jest.fn()}
                onChangeSort={jest.fn()}
                onChangeFolder={jest.fn()}
                onEditAsset={jest.fn()}
                onSelectAllAsset={jest.fn()}
                onSelectAsset={jest.fn()}
                pagination={{ pageCount: 1 }}
                queryObject={{ page: 1, pageSize: 10, filters: { $and: [] } }}
                selectedAssets={[]}
                {...props}
              />
            </NotificationsProvider>
          </IntlProvider>
        </MemoryRouter>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

const setup = (props) => render(<ComponentFixture {...props} />);
describe('BrowseStep', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders and match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should not fetch folder if the user does not have the permission', () => {
    const spy = jest.fn().mockReturnValueOnce({ isLoading: false });
    useFolder.mockImplementationOnce(spy);

    setup({
      canRead: false,
      queryObject: { folder: 1, page: 1, pageSize: 10, filters: { $and: [] } },
    });

    expect(spy).toHaveBeenCalledWith(1, { enabled: false });
  });

  it('should show breadcrumbs navigation', () => {
    setup();

    expect(screen.queryByLabelText('Folders navigation')).toBeInTheDocument();
  });

  it('should hide breadcrumbs navigation if in root folder', () => {
    useFolder.mockReturnValueOnce({ isLoading: false, data: undefined });
    setup();

    expect(screen.queryByLabelText('Folders navigation')).not.toBeInTheDocument();
  });

  it('calls onAddAsset callback', () => {
    const spy = jest.fn();
    const { getByText } = setup({ onAddAsset: spy, folders: [] });
    fireEvent.click(getByText('Add new assets'));
    expect(spy).toHaveBeenCalled();
  });

  it('calls onChangeFolder callback', () => {
    const spy = jest.fn();
    const { getByRole } = setup({ onChangeFolder: spy });
    fireEvent.click(
      getByRole('button', {
        name: /folder 1 : 1 folder, 1 asset/i,
      })
    );
    expect(spy).toHaveBeenCalled();
  });

  it('does display empty state upload first assets if no folder or assets', () => {
    setup({ folders: [], assets: [] });
    expect(screen.getByText('Upload your first assets...')).toBeInTheDocument();
  });

  it('does display empty state no results found if searching with no results', () => {
    setup({
      folders: [],
      assets: [],
      queryObject: { page: 1, pageSize: 10, filters: { $and: [] }, _q: 'true' },
    });
    expect(screen.getByText('There are no assets with the applied filters')).toBeInTheDocument();
  });

  it('does display filters, even if no assets or folders were found', () => {
    setup({
      folders: [],
      assets: [],
      queryObject: { page: 1, pageSize: 10, filters: { $and: [{ mime: 'audio' }] }, _q: '' },
    });
    expect(screen.getByText('Filters')).toBeInTheDocument();
  });

  it('does not display assets title if searching and no folders', () => {
    setup({
      folders: [],
      assets: FIXTURE_ASSETS,
      queryObject: { page: 1, pageSize: 10, filters: { $and: [] }, _q: 'true' },
    });
    expect(screen.queryByText('Assets')).not.toBeInTheDocument();
  });

  it('does not display folders title if searching and no assets', () => {
    setup({
      queryObject: { page: 1, pageSize: 10, filters: { $and: [] }, _q: 'true' },
    });
    expect(screen.queryByText('Folders')).not.toBeInTheDocument();
  });

  it('displays assets and folders titles when there are folders and assets', () => {
    setup({
      assets: FIXTURE_ASSETS,
    });

    expect(screen.getByText('Folders (1)')).toBeInTheDocument();
    expect(screen.getByText('Assets (1)')).toBeInTheDocument();
  });
});
