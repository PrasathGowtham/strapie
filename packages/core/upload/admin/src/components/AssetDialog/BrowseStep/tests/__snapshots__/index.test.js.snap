// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseStep renders and match snapshot 1`] = `
.c68 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c16 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c8 {
  padding-left: 8px;
}

.c10 {
  padding-right: 8px;
}

.c17 {
  padding-top: 12px;
}

.c20 {
  padding-top: 4px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
}

.c22 {
  padding-right: 4px;
  padding-left: 4px;
}

.c25 {
  padding-top: 8px;
  padding-bottom: 8px;
}

.c29 {
  position: relative;
}

.c31 {
  background: #ffffff;
  padding: 12px;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  cursor: pointer;
  cursor: pointer;
}

.c33 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
  border-radius: 4px;
}

.c35 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c38 {
  padding: 4px;
  max-width: 100%;
}

.c40 {
  max-width: 100%;
}

.c46 {
  right: 16px;
}

.c48 {
  padding-top: 16px;
}

.c55 {
  padding-right: 16px;
  padding-left: 16px;
}

.c57 {
  padding-left: 12px;
}

.c1 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c19 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c36 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c41 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c49 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c50 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  line-height: 1.14;
  color: #32324d;
}

.c21 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #32324d;
}

.c23 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #8e8ea9;
}

.c26 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25;
  font-weight: 500;
  color: #32324d;
}

.c42 {
  font-size: 0.875rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
  color: #32324d;
}

.c44 {
  font-size: 0.75rem;
  line-height: 1.33;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #666687;
}

.c56 {
  font-size: 0.875rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c60 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #666687;
}

.c66 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c51 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c32 > * {
  margin-left: 0;
  margin-right: 0;
}

.c32 > * + * {
  margin-left: 8px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c4 svg {
  height: 12px;
  width: 12px;
}

.c4 svg > g,
.c4 svg path {
  fill: #ffffff;
}

.c4[aria-disabled='true'] {
  pointer-events: none;
}

.c4:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c4:focus-visible {
  outline: none;
}

.c4:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c11 {
  height: 100%;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4945ff;
  border: 1px solid #4945ff;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c5 .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 .c6 {
  color: #ffffff;
}

.c5[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c5[aria-disabled='true'] .c6 {
  color: #666687;
}

.c5[aria-disabled='true'] svg > g,.c5[aria-disabled='true'] svg path {
  fill: #666687;
}

.c5[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c5[aria-disabled='true']:active .c6 {
  color: #666687;
}

.c5[aria-disabled='true']:active svg > g,.c5[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c5:hover {
  background-color: #f6f6f9;
}

.c5:active {
  background-color: #eaeaef;
}

.c5 .c6 {
  color: #32324d;
}

.c5 svg > g,
.c5 svg path {
  fill: #32324d;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
}

.c15 svg > g,
.c15 svg path {
  fill: #8e8ea9;
}

.c15:hover svg > g,
.c15:hover svg path {
  fill: #666687;
}

.c15:active svg > g,
.c15:active svg path {
  fill: #a5a5ba;
}

.c15[aria-disabled='true'] {
  background-color: #eaeaef;
}

.c15[aria-disabled='true'] svg path {
  fill: #666687;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 svg {
  height: 4px;
  width: 6px;
}

.c53 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  width: 100%;
  background: transparent;
  border: none;
}

.c53:focus {
  outline: none;
}

.c53[aria-disabled='true'] {
  cursor: not-allowed;
}

.c52 {
  position: relative;
  border: 1px solid #dcdce4;
  padding-right: 12px;
  border-radius: 4px;
  background: #ffffff;
  overflow: hidden;
  min-height: 2rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c52:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c58 {
  background: transparent;
  border: none;
  position: relative;
  z-index: 1;
}

.c58 svg {
  height: 0.6875rem;
  width: 0.6875rem;
}

.c58 svg path {
  fill: #666687;
}

.c59 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: none;
  border: none;
}

.c59 svg {
  width: 0.375rem;
}

.c54 {
  width: 100%;
}

.c27 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 16px;
}

.c28 {
  grid-column: span 3;
  max-width: 100%;
}

.c47 {
  position: absolute;
  top: 12px;
}

.c18:first-child {
  margin-left: calc(-1*8px);
}

.c24 {
  border-radius: 4px;
  color: #666687;
  font-size: 0.75rem;
  line-height: 1.43;
  padding: 4px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c24:hover,
.c24:focus {
  background-color: #dcdce4;
  color: #4a4a6a;
}

.c30 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c30:hover,
.c30:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c34 path {
  fill: currentColor;
}

.c45 {
  display: none;
}

.c37 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c39:focus {
  outline: 2px solid #4945ff;
  outline-offset: -2px;
}

.c61 > * + * {
  margin-left: 4px;
}

.c67 {
  line-height: revert;
}

.c62 {
  padding: 12px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c62:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c62:focus-visible {
  outline: none;
}

.c62:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c64 {
  padding: 12px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c64:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c64:focus-visible {
  outline: none;
}

.c64:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c65 {
  color: #271fe0;
  background: #ffffff;
}

.c65:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c63 {
  font-size: 0.7rem;
  pointer-events: none;
}

.c63 svg path {
  fill: #c0c0cf;
}

.c63:focus svg path,
.c63:hover svg path {
  fill: #c0c0cf;
}

.c3 > * + * {
  margin-left: 8px;
}

.c13 {
  margin-left: auto;
}

.c13 > * + * {
  margin-left: 8px;
}

.c14 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c43 {
  max-width: 100%;
}

@media (max-width:68.75rem) {
  .c28 {
    grid-column: span;
  }
}

@media (max-width:34.375rem) {
  .c28 {
    grid-column: span;
  }
}

<div>
  <div
    class="c0 "
  >
    <div
      class="c0 "
    >
      <div
        class="c0 c1"
      >
        <div
          class="c0 c2 c3"
          wrap="wrap"
        >
          <div>
            <button
              aria-controls="simplemenu-1"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              class="c4 c5"
              label="Sort by"
              type="button"
            >
              <span
                class="c6 c7"
              >
                Sort by
              </span>
              <div
                aria-hidden="true"
                class="c0 c8"
              >
                <span
                  class="c9"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    height="1em"
                    viewBox="0 0 14 8"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                      fill="#32324D"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
            </button>
          </div>
          <button
            aria-disabled="false"
            class="c4 c5"
            type="button"
          >
            <div
              aria-hidden="true"
              class="c0 c10 c11"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M0 4a2 2 0 012-2h20a2 2 0 110 4H2a2 2 0 01-2-2zm4 8a2 2 0 012-2h12a2 2 0 110 4H6a2 2 0 01-2-2zm6 6a2 2 0 100 4h4a2 2 0 100-4h-4z"
                  fill="#32324D"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <span
              class="c6 c7"
            >
              Filters
            </span>
          </button>
        </div>
        <div
          class="c0 c12 c13 c14"
        >
          <span>
            <button
              aria-disabled="false"
              aria-labelledby="tooltip-2"
              class="c4 c15"
              tabindex="0"
              type="button"
            >
              <span
                class="c16"
              >
                Search
              </span>
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M23.813 20.163l-5.3-5.367a9.792 9.792 0 001.312-4.867C19.825 4.455 15.375 0 9.913 0 4.45 0 0 4.455 0 9.929c0 5.473 4.45 9.928 9.912 9.928a9.757 9.757 0 005.007-1.4l5.275 5.35a.634.634 0 00.913 0l2.706-2.737a.641.641 0 000-.907zM9.91 3.867c3.338 0 6.05 2.718 6.05 6.061s-2.712 6.061-6.05 6.061c-3.337 0-6.05-2.718-6.05-6.06 0-3.344 2.713-6.062 6.05-6.062z"
                  fill="#32324D"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="c0 c17"
    >
      <nav
        aria-label="Folders navigation"
        class="c0 "
      >
        <ol
          class="c0 c12 c18"
        >
          <li
            class="c0 c19"
          >
            <div
              class="c0 c20"
            >
              <span
                aria-current="false"
                class="c6 c21"
              >
                Media Library
              </span>
            </div>
            <div
              aria-hidden="true"
              class="c0 c22"
            >
              <span
                class="c6 c23"
              >
                /
              </span>
            </div>
          </li>
          <li
            class="c0 c19"
          >
            <button
              class="c24"
              type="button"
            >
              Folder 2
            </button>
            <div
              aria-hidden="true"
              class="c0 c22"
            >
              <span
                class="c6 c23"
              >
                /
              </span>
            </div>
          </li>
          <li
            class="c0 c19"
          >
            <button
              class="c24"
              type="button"
            >
              Folder 1
            </button>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="c0 "
    >
      <div
        class="c0 c25"
      >
        <h2
          class="c6 c26"
        >
          Folders (1)
        </h2>
      </div>
      <div
        class="c0 c27"
      >
        <div
          class="c28"
        >
          <div
            class="c0 "
          >
            <div
              class="c0 c29 "
              tabindex="0"
            >
              <button
                aria-hidden="true"
                aria-label="Folder 1"
                class="c30"
                tabindex="-1"
                type="button"
              />
              <div
                class="c0 c31 c12 c32"
                cursor="pointer"
                spacing="2"
              >
                <div
                  class="c0 c33"
                >
                  <svg
                    class="c34"
                    fill="none"
                    height="1.5rem"
                    viewBox="0 0 24 24"
                    width="1.5rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.414 5H21a1 1 0 011 1v14a1 1 0 01-1 1H3a1 1 0 01-1-1V4a1 1 0 011-1h7.414l2 2z"
                      fill="#212134"
                    />
                  </svg>
                </div>
                <div
                  class="c0 c35 c36 c37"
                  id="folder-1-1-title"
                  overflow="hidden"
                >
                  <button
                    class="c0 c38 c39"
                    type="button"
                  >
                    <h2
                      class="c0 c40 c41"
                    >
                      <span
                        class="c6 c42 c43"
                      >
                        Folder 1
                        <div
                          class="c16"
                        >
                          :
                        </div>
                      </span>
                      <span
                        class="c6 c44 c43"
                      >
                        1 folder, 1 asset
                      </span>
                    </h2>
                  </button>
                </div>
                <div
                  class="c0 c45"
                >
                  <div
                    class="c0 c46 c12 c32 c47"
                    spacing="2"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0 c48 c49"
    >
      <div
        class="c0 c12"
      >
        <div>
          <div
            class="c0 c50 c51"
          >
            <div
              class="c0 c12 c52"
            >
              <button
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Entries per page"
                aria-labelledby="select-4-label select-4-content"
                class="c53"
                id="select-4"
                type="button"
              />
              <div
                class="c0 c49 c54"
              >
                <div
                  class="c0 c12"
                >
                  <div
                    class="c0 c55"
                  >
                    <span
                      class="c6 c56"
                      id="select-4-content"
                    >
                      10
                    </span>
                  </div>
                </div>
                <div
                  class="c0 c12"
                >
                  <button
                    aria-hidden="true"
                    class="c0 c57 c58 c59"
                    tabindex="-1"
                    title="Carret Down Button"
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      viewBox="0 0 14 8"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                        fill="#32324D"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c0 c8"
        >
          <label
            class="c6 c60"
            for="page-size"
          >
            Entries per page
          </label>
        </div>
      </div>
      <nav
        aria-label="pagination"
        class=""
      >
        <ul
          class="c0 c12 c61"
        >
          <li>
            <button
              aria-disabled="true"
              class="c62 c63"
              tabindex="-1"
              type="button"
            >
              <div
                class="c16"
              >
                Go to previous page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.88 14.12L3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88z"
                  fill="#32324D"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              class="c64 c65"
              type="button"
            >
              <div
                class="c16"
              >
                Go to page 1
              </div>
              <span
                aria-hidden="true"
                class="c6 c66 c67"
              >
                1
              </span>
            </button>
          </li>
          <li>
            <button
              aria-disabled="true"
              class="c62 c63"
              tabindex="-1"
              type="button"
            >
              <div
                class="c16"
              >
                Go to next page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1.88L6.107 8 0 14.12 1.88 16l8-8-8-8L0 1.88z"
                  fill="#32324D"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    class="c68"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
